lib/metrifox_sdk/usages/api.rb

```diff
@@ -29,13 +29,23 @@ def fetch_usage(base_url, api_key, request_payload)
       customer_key = get_value(request_payload, :customer_key)
       event_name = get_value(request_payload, :event_name)
       amount = get_value(request_payload, :amount) || 1
+      credit_used = get_value(request_payload, :credit_used)
+      event_id = get_value(request_payload, :event_id)
+      timestamp = get_value(request_payload, :timestamp)
+      metadata = get_value(request_payload, :metadata) || {}
 
       body = {
         customer_key: customer_key,
         event_name: event_name,
         amount: amount
       }
 
+      # Add optional fields if present
+      body[:credit_used] = credit_used if credit_used
+      body[:event_id] = event_id if event_id && !event_id.empty?
+      body[:timestamp] = timestamp if timestamp
+      body[:metadata] = metadata if metadata && !metadata.empty?
```

Comment:
We assign metadat to `{}` if not added, but here, we reject it if empty. `{}.empty? == true`